### DISCUSSION TITLE
Element histories

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/cms#8638] [Ideas]
Craft should store a history for each element. Who created it, who modified it, who deleted it, etc.

Would resolve #3802 and #3945.

**Comment by timkelty** [comment]
Similarly – 

It can be hard to rely on fields like `entries.dateUpdated` (e.g. displaying "Post last edited on [date]"), as they get can updated programmatically, like when you re-save a section or field layout.

Rather than trying to use those fields, presumably we'd be able to do something like:

```
Post last edited on {{ element.history.one().editDate|date }}
```

…presuming that programatic saves didn't save to the history.

**Comment by jsunsawyer** [comment]
@timkelty I was just about to add an issue related to that.

Programatic saves are making sorting entries in the CP by `Date Updated` pretty much useless.

**Comment by ademers** [comment]
This can't come soon enough! Thanks!

I created and maintain a membership site where several Users can modify User accounts. It would be very useful and helpful to be able to see, in the CP, who updated which User account and when.

- Last Updated: (date)
- Updated By: (user account)

**Comment by bencroker** [comment]
Element changes are already being tracked in the `changedattributes` and `changedfields` tables in Craft 4. It seems like all that is left to do is to switch from using `UPSERT` to `INSERT` and build the UI for the history view.
